src/Configuration/Compiler/TaskCompilerPass.php

```diff
@@ -45,7 +45,7 @@ public function process(ContainerBuilder $container): void
             // Determine Keys:
             $currentTaskService = $availableTasks[$currentTaskName];
             ['id' => $taskId, 'class' => $taskClass,] = $currentTaskService;
-            $configuredTaskKey = $taskId.'.configured';
+            $configuredTaskKey = $taskName.'.configured';
```

Comment:
Maybe better to name it `taskId.taskName.configured`?
Just in case to make it clear which task is configured if somebody creates e.g. a phpunit alias for a shell task?

Reply:
Good idea I added it!